devicetypes/smartthings/zll-dimmer-bulb.src/zll-dimmer-bulb.groovy

```diff
@@ -96,7 +97,35 @@ def poll() {
     refresh()
 }
 
-def configure() {
-    log.debug "Configuring Reporting and Bindings."
+/**
+ * PING is used by Device-Watch in attempt to reach the Device
+ * */
+def ping() {
+    return zigbee.levelRefresh()
+}
+
+def healthPoll() {
+    log.debug "healthPoll()"
+    def cmds = refresh()
+    cmds.each{ sendHubCommand(new physicalgraph.device.HubAction(it))}
+}
+
+def configureHealthCheck() {
+    Integer hcIntervalMinutes = 12
+    log.debug "Configuring Health Check, Reporting and Bindings."
+    unschedule("healthPoll")
+    runEvery5Minutes("healthPoll")
+    // Device-Watch allows 2 check-in misses from device
+    sendEvent(name: "checkInterval", value: 60 * hcIntervalMinutes, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
     zigbee.onOffConfig() + zigbee.levelConfig() + zigbee.onOffRefresh() + zigbee.levelRefresh()
 }
+
+def configure() {
+    log.debug "configure()"
+    configureHealthCheck()
```

Comment:
Necessary? I'm not sure. This was a pattern I'd seen in the docs, and now that I look, it was  used in SmartApps calling a common initialize() method from both configure() and updated():
http://docs.smartthings.com/en/latest/smartapp-developers-guide/anatomy-and-life-cycle-of-a-smartapp.html#smartapp-structure

Looking at your comment on CHF-429, it seems that updated() is called if the DH for a device is changed in the IDE. What about in the case of deploying a new version of the **same** DH which changes the checkInterval?

Regardless, the best answer I can give is another question: For a DH which implements capability configurable, can configure() ever be invoked in an execution where updated() would not be?

If the answer is a definitive no, then there's no need to call it from both places.

Finally, it seems that I should leave the ZigBee.onOffConfig()... line with configure() for these DHs as that's configuring the device, which is the intent of the capability.configurable - do you agree?